src/routes/transactions/helpers/kiln-native-staking.helper.ts

```diff
@@ -23,38 +23,12 @@ export class KilnNativeStakingHelper {
     to: `0x${string}`;
     data: `0x${string}`;
   } | null> {
-    const selector = toFunctionSelector('function deposit() external payable');
-    const transaction = this.transactionFinder.findTransaction(
-      (transaction) => transaction.data.startsWith(selector),
-      args,
-    );
-
-    if (!transaction?.to) {
-      return null;
-    }
-
-    // We need to check against the deployment as `deposit` is a common function name
-    const deployment = await this.stakingRepository
-      .getDeployment({
-        chainId: args.chainId,
-        address: transaction.to,
-      })
-      .catch(() => null);
-
-    if (
-      deployment?.product_type !== 'dedicated' &&
-      deployment?.chain !== 'unknown'
-    ) {
-      return null;
-    }
-
-    return {
-      to: transaction.to,
-      data: transaction.data,
-    };
+    return this.findTransactionAndCheckDeployment({
+      ...args,
+      selector: toFunctionSelector('function deposit() external payable'),
```

Comment:
We can move `toFunctionSelector` into `findTransactionAndCheckDeployment` as well. Perhaps we should also then move the functions to constants thereafter too. What do you think?

Reply:
I've added your suggestion and I've gone a bit even further in the refactor by introducing a `findTransaction` function which calls `checkDeployment`. Please let me know if you think it doesn't make sense, but I think the complexity has decreased: [deb0fcf](https://github.com/safe-global/safe-client-gateway/pull/1895/commits/deb0fcff225ab0d8949806901c165c62c814e56d)

Reply:
After an internal sync, I think we've agreed to expose a function per transaction type instead of maintaining a mapping object. I've changed the approach in [c1ad65d](https://github.com/safe-global/safe-client-gateway/pull/1895/commits/c1ad65d95e9dfab881d59be4c56e1ea9185b5e3d)

Reply:
And I've also moved the signatures to class constants in [1c4ebea](https://github.com/safe-global/safe-client-gateway/pull/1895/commits/1c4ebea1918019d6a0572689a388b4e52f350e0f)